packages/store/addon/-private/system/core-store.ts

```diff
@@ -285,6 +286,29 @@ abstract class CoreStore extends Service {
     }
 
     if (DEBUG) {
+      // support for moduleFor style unit tests
+      // that were relying on ember-test-helpers
+      // doing an auto-registration of the transform
+      // or us doing one
+      const Mapping = {
+        date: 'DateTransform',
+        boolean: 'BooleanTransform',
+        number: 'NumberTransform',
+        string: 'StringTransform',
+      };
+
+      Object.keys(Mapping).forEach(attributeType => {
+        const transform = getOwner(this).lookup(`transform:${attributeType}`);
+
+        if (!transform) {
+          // we don't deprecate this because the moduleFor style tests with the closed
```

Comment:
IMHO, we should probably emit a deprecation in this path still. `moduleFor` itself is deprecated already but who knows how long that will last. We could do it as a private API deprecation for removal in ~ 3.17.0 (I'd hate to have to support this forever...). The work around for folks still wanting `moduleFor` is to add `integration: true` and remove `needs`, so its overall much smaller work than migrating the whole suite.

Reply:
When we discussed this a while back we felt deprecating here and deprecating in ember-test-helpers would create unnecessary noise. Has your thinking changed?

Reply:
discussed oob, will add deprecations

Reply:
added